Machines/HPSS/arconfig_wrfout_fineIO_2.sb

```diff
@@ -0,0 +1,1006 @@
+          if [[ ${word_tf} == ${word} ]]; then
+            EXIST="EXIST" 
+          fi
+        done
+      fi  
+      if [[ -n ${EXIST} ]]; then
+        SNOW_D1_CHECKED="${SNOW_D1_CHECKED} ${word}"
+      fi  
+    done  
+    if [[ -n ${SNOW_D1_CHECKED} ]]; then
+      SNOW_D1=${SNOW_D1_CHECKED}    
+    else
+      unset SNOW_D1
+      echo 'SNOW_D1 assigned files did not exist. SNOW_D1 was unset.'
+    fi 
+    local SNOW_D2_CHECKED=""
+    if [[ ${MODE} == 'RETRIEVE' ]]; then    
+      local TAR_FILES=$( htar -tf ${DST}"/wrfsnow_d02_${DATENAME}.tar" 2>/dev/null | awk '{print $7}' | head -n -2 )
+    fi     
+    for word in ${SNOW_D2}; do
+      if [[ ${MODE} == 'BACKUP' ]]; then
+        local EXIST=$( ls ${word} 2>/dev/null )
+      elif [[ ${MODE} == 'RETRIEVE' ]]; then
+        unset EXIST
+        for word_tf in ${TAR_FILES}; do
+          if [[ ${word_tf} == ${word} ]]; then
+            EXIST="EXIST" 
+          fi
+        done
+      fi  
+      if [[ -n ${EXIST} ]]; then
+        SNOW_D2_CHECKED="${SNOW_D2_CHECKED} ${word}"
+      fi  
+    done  
+    if [[ -n ${SNOW_D2_CHECKED} ]]; then
+      SNOW_D2=${SNOW_D2_CHECKED}    
+    else
+      unset SNOW_D2
+      echo 'SNOW_D2 assigned files did not exist. SNOW_D2 was unset.'
+    fi
+    local SNOW_D3_CHECKED=""
+    if [[ ${MODE} == 'RETRIEVE' ]]; then    
+      local TAR_FILES=$( htar -tf ${DST}"/wrfsnow_d03_${DATENAME}.tar" 2>/dev/null | awk '{print $7}' | head -n -2 )
+    fi     
+    for word in ${SNOW_D3}; do
+      if [[ ${MODE} == 'BACKUP' ]]; then
+        local EXIST=$( ls ${word} 2>/dev/null )
+      elif [[ ${MODE} == 'RETRIEVE' ]]; then
+        unset EXIST
+        for word_tf in ${TAR_FILES}; do
+          if [[ ${word_tf} == ${word} ]]; then
+            EXIST="EXIST" 
+          fi
+        done
+      fi  
+      if [[ -n ${EXIST} ]]; then
+        SNOW_D3_CHECKED="${SNOW_D3_CHECKED} ${word}"
+      fi  
+    done  
+    if [[ -n ${SNOW_D3_CHECKED} ]]; then
+      SNOW_D3=${SNOW_D3_CHECKED}    
+    else
+      unset SNOW_D3
+      echo 'SNOW_D3 assigned files did not exist. SNOW_D3 was unset.'
+    fi
+    local SNOW_D4_CHECKED=""
+    if [[ ${MODE} == 'RETRIEVE' ]]; then    
+      local TAR_FILES=$( htar -tf ${DST}"/wrfsnow_d04_${DATENAME}.tar" 2>/dev/null | awk '{print $7}' | head -n -2 )
+    fi     
+    for word in ${SNOW_D4}; do
+      if [[ ${MODE} == 'BACKUP' ]]; then
+        local EXIST=$( ls ${word} 2>/dev/null )
+      elif [[ ${MODE} == 'RETRIEVE' ]]; then
+        unset EXIST
+        for word_tf in ${TAR_FILES}; do
+          if [[ ${word_tf} == ${word} ]]; then
+            EXIST="EXIST" 
+          fi
+        done
+      fi  
+      if [[ -n ${EXIST} ]]; then
+        SNOW_D4_CHECKED="${SNOW_D4_CHECKED} ${word}"
+      fi  
+    done  
+    if [[ -n ${SNOW_D4_CHECKED} ]]; then
+      SNOW_D4=${SNOW_D4_CHECKED}    
+    else
+      unset SNOW_D4
+      echo 'SNOW_D4 assigned files did not exist. SNOW_D4 was unset.'
+    fi    
```

Comment:
So much boiler-plate code... I wonder if there would have been a more elegant way to do this... I don't know, though